specutils/analysis/width.py

```diff
@@ -215,8 +231,31 @@ def _compute_gaussian_sigma_width(spectrum, regions=None):
         centroid_result = centroid_result[:, np.newaxis]
 
     dx = (spectral_axis - centroid_result)
-    sigma = np.sqrt(np.sum((dx * dx) * flux, axis=-1) / np.sum(flux, axis=-1))
+    numerator = np.sum((dx * dx) * flux, axis=-1)
+    denom = np.sum(flux, axis=-1)
+    sigma2 = numerator / denom
+    sigma = np.sqrt(sigma2)
+    if centroid_result.uncertainty is not None:
+        # NOTE: until/unless disp_uncert is supported, dx_uncert == centroid_result.uncertainty
+        disp_uncert = 0.0 * spectral_axis.unit
+        dx_uncert = np.sqrt(disp_uncert**2 + centroid_result.uncertainty**2)
+        # dx_uncert = centroid_result.uncertainty
+
+        # uncertainty for each term in the numerator sum
+        num_term_uncerts = dx * dx * flux * np.sqrt(2*(dx_uncert/dx)**2 + (flux_uncert/flux)**2)
+        # uncertainty (squared) for the numerator, added in quadrature
+        num_uncertsq = np.sum(num_term_uncerts**2, axis=-1)
+        # uncertainty (squared) for the denomenator
+        denom_uncertsq = np.sum(flux_uncert**2)
+
+        sigma2_uncert = numerator/denom * np.sqrt(num_uncertsq * numerator**-2 +
+                                                  denom_uncertsq * denom**-2)
+
+        sigma.uncertainty = 0.5 * sigma2_uncert / sigma2 * sigma.unit
```

Comment:
it would be helpful to write out the math behind this in an associated latex document

Reply:
I've checked by hand and this all looks right to me though

Reply:
This is a good idea (even though this should just follow normal uncertainty propagation rules, the code can definitely be difficult to parse, even with splitting into "steps" to try to keep it readable).  I think this probably should be done for all analysis functions that expose uncertainties in some consolidated place in the docs, and that might be better suited for a follow-up effort since its more broad?

If you/anyone thinks it should be included before merging this, let me know where it belongs and I can try to write it up and include it.

Reply:
Followup is fine by me.